src/store/movies-store/movies.slice.ts

```diff
@@ -0,0 +1,79 @@
+import {createSlice, PayloadAction} from "@reduxjs/toolkit";
+import {Movie} from "../../api/movies-api/models/movie.model";
+import {Pagination} from "../../api/shared/models/pagination.model";
+import {PaginatedResult} from "../../api/shared/models/paginated-result.model";
+
+export type MoviesState = {
+  entities: Movie[];
+  loading: boolean;
+  pagination: Pagination;
+}
+
+const initialState: MoviesState = {
+  entities: [],
+  loading: false,
+  pagination: {
+    limit: 0,
+    offset: 0
+  }
+}
+
+export const moviesSlice = createSlice({
+  name: 'Movies Slice',
+  initialState,
+  reducers: {
+    getMovies: (state: MoviesState, action: PayloadAction<Pagination>) => {
+        state.entities
+    },
+    getMoviesSuccess: (state: MoviesState, action: PayloadAction<PaginatedResult<Movie>>) => {
+      return {
+        ...state,
+        loading: false,
+        pagination: {
+          ...state.pagination,
+          limit: action.payload.limit,
```

Comment:
Please, use destructuring. Check all occurrences

Reply:
what do you mean, there are already destructuring everywhere?